plugins/modules/foreman_scc_account.py

```diff
@@ -0,0 +1,170 @@
+    required: false
+    type: str
+    choices: ["never", "daily", "weekly", "monthly"]
+  sync_date:
+    description: Last Sync time of scc_account
+    required: false
+    type: str
+  organization:
+    description: Name of related organization
+    type: str
+  test_connection:
+    description: Test connection of scc_account
+    required: false
+    default: false
+    type: bool
+  updated_name:
+    description: Name to be updated of scc_account
+    type: str
+  state:
+    description: State of the scc_account
+    default: present
+    choices: ["present", "absent", "synced"]
+    type: str
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: "Create an scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test"
+    login: "abcde"
+    scc_account_password: "12345"
+    base_url: "https://scc.suse.com"
+    state: present
+
+- name: "Update a scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test1"
+    state: present
+
+- name: "Delete a scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.foreman_helper import KatelloEntityAnsibleModule
+
+
+def main():
+    module = KatelloEntityAnsibleModule(
+        entity_spec=dict(
+            name=dict(required=True),
+            updated_name=dict(),
+            login=dict(),
+            scc_account_password=dict(no_log=True, flat_name='password'),
+            base_url=dict(),
+            sync_date=dict(),
+            interval=dict(choices=['never', 'daily', 'weekly', 'monthly']),
+            state=dict(default='present', choices=['present', 'absent', 'synced']),
+        ),
+        argument_spec=dict(
+            test_connection=dict(type='bool', default=False))
+    )
+
+    entity_dict = module.clean_params()
+
+    module.connect()
+
+    entity_dict['organization'] = module.find_resource_by_name('organizations', entity_dict['organization'], thin=True)
+
+    scope = {'organization_id': entity_dict['organization']['id']}
+
+    entity = module.find_resource_by_name('scc_accounts', name=entity_dict['name'], params=scope, failsafe=True)
+
+    if entity_dict['test_connection']:
+        if not entity and not module.desired_absent:
+            scc_account_credentials = {'login': entity_dict['login'], 'password': entity_dict['scc_account_password'], 'base_url': entity_dict['base_url']}
+            connected = module.foremanapi.resource('scc_accounts').call('test_connection', scc_account_credentials)
+        elif module.desired_absent:
+            connected = True
+        else:
+            connected = module.foremanapi.resource('scc_accounts').call('test_connection', {'id': entity['id']})
```

Comment:
This should test the existing account with overwritten credentials (as much as they are provided) to make sure, the state we want to put the account into can do the connection.